### PR TITLE
feat(replay): Update privacy docs for clarity

### PR DESCRIPTION
**Pull request by mydea**
* Clarifies that masking happens in the SDK
* Updates the types for the replay options to be code
* Adds config to the mask/block/ignore sections directly

Closes https://github.com/getsentry/sentry-docs/issues/8517

### CODE REVIEW

File: src/platform-includes/replay/privacy-configuration/javascript.mdx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-| unmask        | string[]                 | `.sentry-unmask, [data-sentry-unmask]`   | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked with `maskAllText`.                                                                      |
+| key           | type               | default                                  | description                                                                                                                                                                                       |
+| ------------- | ------------------ | ---------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| mask          | `string[]`         | `.sentry-mask, [data-sentry-mask]`       | Mask all elements that match the given DOM selectors. See [Masking](#masking) section for an example. Note that any configured selectors will be in _addition_ to the defaults.                   |
```
Comment (lforst) [inline src/platform-includes/replay/privacy-configuration/javascript.mdx:5]: Can we make this also valid js (i.e. strings)? Same for the `block`, `unblock`, `ignore` and `unmask` variables below.

File: src/platform-includes/replay/privacy-configuration/javascript.mdx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-| block         | string[]                 | `.sentry-block, [data-sentry-block]`     | Redact all elements that match the DOM selector(s). See [Blocking](#blocking) section for an example. Note that any configured selectors will be in _addition_ to the defaults.                   |
-| blockAllMedia | boolean                  | `true`                                   | Block _all_ media elements (`img`, `svg`, `video`, `object`, `picture`, `embed`, `map`, `audio`).                                                                                                 |
-| ignore        | string[]                 | `.sentry-ignore, [data-sentry-ignore]`   | Ignores all events on the matching input fields. See [Ignoring](#ignoring) above for an example.                                                                                                  |
-| maskFn        | (text: string) => string | `(text) => '*'.repeat(text.length)`      | Function to customize how text content is masked before sending to server. By default, masks text with `*`.                                                                                       |
```
Comment (lforst) [inline src/platform-includes/replay/privacy-configuration/javascript.mdx:11]: Why remove the `: string` here?

**Review by lforst — APPROVED** [review]
❤️

**Review by billyvg — APPROVED** [review]
lgtm aside from luca's comments

**Review by shanamatthews — APPROVED** [review]
Fine from my perspective!

File: src/platform-includes/replay/privacy-configuration/javascript.mdx (hunk 2 above)
Comment (lforst) [reply src/platform-includes/replay/privacy-configuration/javascript.mdx:11]: If you ask me we should prioritize correctness over looks here 😄 But I'll leave it up to you.

**Pull request merged**